cosmos/scheduler/cmd/scheduler.go

```diff
@@ -37,6 +39,7 @@ func (s *Scheduler) Run() {
 		log.Println(err)
 		return
 	}
+	go s.checkMongoConnection()
```

Comment:
Instead of manually managing timers and crashing, have you thought about implementing a tiny `/healthcheck` route that does the ping, and can be called by kube directly? This makes it easy to have "flexible" behavior (such as interval, initial delay, thresholds, ...) at no cost.

Reply:
I have considered that as part of the larger refactor that I'm working on. But I'll add this route instead of having temporary solutions.